File: medplum-java/medplum-model/src/main/java/com/fhir/QuestionnaireResponse.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableQuestionnaireResponse.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface QuestionnaireResponse {
  Optional<QuestionnaireresponseStatus> status();

  Optional<List<Extension>> extension();

  Optional<List<Reference>> partOf();

  Optional<Reference> subject();

  Optional<Identifier> identifier();

  Optional<Reference> encounter();

  Optional<Reference> author();

  String resourceType();

  Optional<uri> implicitRules();

  Optional<Reference> source();

  Optional<Narrative> text();

  Optional<code> language();

  Optional<canonical> questionnaire();

  Optional<List<QuestionnaireResponse_Item>> item();

  Optional<List<Reference>> basedOn();

  Optional<List<Extension>> modifierExtension();

  Optional<dateTime> authored();

  Optional<Meta> meta();

  Optional<List<ResourceList>> contained();

  Optional<id> id();

  static ImmutableQuestionnaireResponse.ResourceTypeBuildStage builder() {
    return ImmutableQuestionnaireResponse.builder();
  }
}

File: medplum-typescript/src/model/fhir/QuestionnaireResponse.ts
import { canonical } from "./canonical";
import { code } from "./code";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { id } from "./id";
import { Identifier } from "./Identifier";
import { Meta } from "./Meta";
import { Narrative } from "./Narrative";
import { QuestionnaireresponseStatus } from "./QuestionnaireresponseStatus";
import { QuestionnaireResponse_Item } from "./QuestionnaireResponse_Item";
import { Reference } from "./Reference";
import { ResourceList } from "./ResourceList";
import { uri } from "./uri";

export interface QuestionnaireResponse {
    resourceType: string;
    id: id | null | undefined;
    meta: Meta | null | undefined;
    implicitRules: uri | null | undefined;
    language: code | null | undefined;
    text: Narrative | null | undefined;
    contained: ResourceList[] | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    identifier: Identifier | null | undefined;
    basedOn: Reference[] | null | undefined;
    partOf: Reference[] | null | undefined;
    questionnaire: canonical | null | undefined;
    status: QuestionnaireresponseStatus | null | undefined;
    subject: Reference | null | undefined;
    encounter: Reference | null | undefined;
    authored: dateTime | null | undefined;
    author: Reference | null | undefined;
    source: Reference | null | undefined;
    item: QuestionnaireResponse_Item[] | null | undefined;
}
